implementations/elixir/ockam/ockam/mix.exs

```diff
@@ -45,8 +45,9 @@ defmodule Ockam.MixProject do
 
   defp deps do
     [
-      {:telemetry, "~> 0.4.2"},
+      {:telemetry, "~> 0.4.2", optional: true},
       {:gen_state_machine, "~> 3.0"},
+      {:ranch, "~> 2.0", optional: true},
       {:ex_doc, "~> 0.23.0", only: :dev, runtime: false},
       {:credo, "~> 1.5", only: [:dev, :test], runtime: false},
       {:dialyxir, "~> 1.0", only: [:dev], runtime: false}
```

Comment:
out of curiosity, could `dialyxir` being `only: [:dev]` miss the chance that an external PR is accepted and breaks some type contract that the tests dont find? e.g. If the PR was made by someone who didn't run the code locally, just observed that tests pass locally or on CI. 

in the case that this happens, the code could be correct (using a value like 1 instead of some equivalent typed enum or otherwise), but the type checking would fail.

Reply:
our CI builds in `dev`, that's the convention when linting elixir projects.
So when we turn on dialyxir in CI, it won't pass CI

Reply:
I feel the need to add that since Elixir is dynamically typed, dialyzir will only do so much ... type errors are the cost of the convenience. That's why https://gleam.run/ is exciting as a tool in the BEAM toolbox.